src/Sampler.cpp

```diff
@@ -0,0 +1,233 @@
+    if (dimension == 2)
+        nu_parameter = 1.0;
+    else
+        nu_parameter = 0.5;
+    double ddim = static_cast<double>(dimension);
+    scalar_g_ = std::pow(4.0 * M_PI, ddim / 4.0) * std::pow(kappa, nu_parameter) *
+                std::sqrt( tgamma(nu_parameter + ddim / 2.0) / tgamma(nu_parameter) );
+}
+
+PDESampler::~PDESampler()
+{
+}
+
+/// @todo cell_volume should be variable rather than constant
+void PDESampler::NewSample()
+{
+    current_state_ = FINE_SAMPLE;
+
+    // construct white noise right-hand side
+    // (cell_volume is supposed to represent fine-grid W_h)
+    for (int i = 0; i < fine_vector_size_; ++i)
+    {
+        rhs_fine_(i) = scalar_g_ * std::sqrt(cell_volume_) *
+                       normal_distribution_.Sample();
+    }
+}
+
+void PDESampler::NewCoarseSample()
+{
+    current_state_ = COARSE_SAMPLE;
+    MFEM_ASSERT(false, "Not implemented!");
+}
+
+mfem::Vector& PDESampler::GetFineCoefficient()
+{
+    MFEM_ASSERT(current_state_ == FINE_SAMPLE,
+                "PDESampler object in wrong state (call NewSample() first)!");
+
+    fvupscale_->SolveFine(rhs_fine_, coefficient_fine_);
+    for (int i = 0; i < coefficient_fine_.Size(); ++i)
+    {
+        coefficient_fine_(i) = std::exp(coefficient_fine_(i));
+    }
+    return coefficient_fine_;
+}
+
+/**
+   For myself:
+
+   c_i              : coefficient for coarse basis function, representing ~normal field K
+   (c_i / q_i)      : value of ~normal field K on agg i
+   exp(c_i/q_i)     : value of lognormal field exp(K) on agg i
+   exp(c_i/q_i) q_i : coefficient for coarse basis function, representing lognormal field exp(K)
+
+   indexing: we consider only the coefficient for the *constant* component i
+*/
+mfem::Vector& PDESampler::GetCoarseCoefficient()
+{
+    MFEM_ASSERT(current_state_ == FINE_SAMPLE ||
+                current_state_ == COARSE_SAMPLE,
+                "PDESampler object in wrong state (call NewSample() first)!");
+
+    if (current_state_ == FINE_SAMPLE)
+        fvupscale_->Restrict(rhs_fine_, rhs_coarse_);
+    mfem::Vector coarse_sol = fvupscale_->GetCoarseVector();
+    fvupscale_->SolveCoarse(rhs_coarse_, coarse_sol);
+    coarse_sol *= -1.0; // ?
+
+    coefficient_coarse_ = 0.0;
+    const mfem::Vector& coarse_constant_rep = fvupscale_->GetGraphCoarsen().GetCoarseConstantRep();
+    MFEM_ASSERT(coarse_constant_rep.Size() == coarse_sol.Size(),
+                "PDESampler::GetCoarseCoefficient : Sizes do not match!");
+    int agg_index = 0;
+    for (int i = 0; i < coarse_sol.Size(); ++i)
+    {
+        if (std::fabs(coarse_constant_rep(i)) > 1.e-8)
+        {
+            coefficient_coarse_(agg_index++) =
+                std::exp(coarse_sol(i) / coarse_constant_rep(i));
+        }
+    }
+    MFEM_ASSERT(agg_index == num_coarse_aggs_, "Something wrong in coarse_constant_rep!");
+
+    return coefficient_coarse_;
+}
+
+/// @todo ugly hack for debugging, @deprecated
```

Comment:
Adding an already deprecated function in a PR doesn't sound like a good idea.
If it is still needed, maybe remove the tag.

Reply:
I removed the tag. I like what this does in `examples/sampler.cpp`, so I think it is worth keeping around.